editor/doc_tools.cpp

```diff
@@ -350,17 +350,20 @@ void DocTools::generate(bool p_basic_types) {
 		List<PropertyInfo> properties;
 		List<PropertyInfo> own_properties;
 		if (name == "ProjectSettings") {
-			//special case for project settings, so settings can be documented
+			// Special case for project settings, so settings can be documented.
 			ProjectSettings::get_singleton()->get_property_list(&properties);
 			own_properties = properties;
 		} else {
 			ClassDB::get_property_list(name, &properties);
 			ClassDB::get_property_list(name, &own_properties, true);
 		}
 
+		properties.sort();
+		own_properties.sort();
+
 		List<PropertyInfo>::Element *EO = own_properties.front();
 		for (const PropertyInfo &E : properties) {
-			bool inherited = EO == nullptr;
+			bool inherited = true;
```

Comment:
What does this change do?

Reply:
So how it worked was that you get two lists, just own properties and all the properties including parent classes. Both are reported in their bind order, similar to inspector. Then we iterate the list of all properties. The assumption is that the list of all properties contains all the own properties first. So by the time `EO` is null, the own properties have ended and it is safe to assume every other property is inherited.

While `EO` is true we compare the values (below) and move the cursor one position forward. This still works when I add sorting, but the caveat is that we can no longer assume that own properties are the first ones. They still follow the same order, just have gaps of inherited properties in between. So we can no longer use that old default value, but its safe to default to true. Every own property will be caught by the check below, as long as there is anything left to check.